src/main/extension.cpp

```diff
@@ -89,11 +91,21 @@ bool VersioningUtils::IsSupportedCAPIVersion(string &capi_version_string) {
 }
 
 bool VersioningUtils::IsSupportedCAPIVersion(idx_t major, idx_t minor, idx_t patch) {
-	if (major > DUCKDB_EXTENSION_API_VERSION_MAJOR || minor > DUCKDB_EXTENSION_API_VERSION_MINOR ||
-	    patch > DUCKDB_EXTENSION_API_VERSION_PATCH) {
+	if (major > DUCKDB_EXTENSION_API_VERSION_MAJOR) {
+		return false;
+	}
+	if (major < DUCKDB_EXTENSION_API_VERSION_MAJOR) {
+		return true;
```

Comment:
```suggestion
	if (major != DUCKDB_EXTENSION_API_VERSION_MAJOR) {
		return false;
```
Discussed on a side, this allows to refuse load for `0.0.1` and we can eventually ship `2.0.0` that is not compatible.